### PR TITLE
SEGSAdmin config and logging functionality

### PR DESCRIPTION
**Pull request by miles200**
* Adds ability to edit and save server and logging settings
* Adds more existing settings.cfg and databases checks
* Initial/prototype user friendly 'server setup' functionality
* User can provide an IP and a default config file is created

Adds to #285  .

### CODE REVIEW

File: Projects/CoX/Utilities/SEGSAdmin/SEGSAdminTool.ui [hunk 1]
```diff
@@ -20,898 +20,6 @@
...
-       <item>
-        <widget class="QToolButton" name="selectfile">
-         <property name="text">
-          <string>...</string>
-         </property>
-        </widget>
-       </item>
-      </layout>
-     </widget>
-     <widget class="QWidget" name="layoutWidget">
```
Comment (ldilley) [inline Projects/CoX/Utilities/SEGSAdmin/SEGSAdminTool.ui]: This use of "layoutWidget" should differ from line 122. Each instance should have a unique name. This includes lines 196 and 915 as well.

**Comment by miles200** [comment]
@nemerle @ldilley thanks for this, these changes should have been pushed up now. 
Bit confused on the layoutWidget review, have dropped you a message in discord

File: Projects/CoX/Utilities/SEGSAdmin/GenerateConfigFileDialog.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    ui->config_server_name_edit->clear();
+    show();
+}
+
+void GenerateConfigFileDialog::capture_input()
+{
+QString server_name = ui->config_server_name_edit->text();
+QString ip = ui->config_ip_edit->text();
+emit sendInputConfigFile(server_name,ip);
+}
```
Comment (nemerle) [inline Projects/CoX/Utilities/SEGSAdmin/GenerateConfigFileDialog.cpp]: This looks like it's missing indent :)

File: Projects/CoX/Utilities/SEGSAdmin/SEGSAdminTool.h [hunk 3]
```diff
@@ -20,29 +20,36 @@ class SEGSAdminTool : public QMainWindow
 {
     Q_OBJECT
     class AddNewUserDialog *m_add_user_dialog;
+    class GenerateConfigFileDialog *m_generate_config_dialog;
 
 public:
     explicit SEGSAdminTool(QWidget *parent = 0);
     ~SEGSAdminTool();
```
Comment (nemerle) [inline Projects/CoX/Utilities/SEGSAdmin/SEGSAdminTool.h]: setting pointer variables to `0`, should be done with `nullptr` instead, the meaning is clearer that way

Comment (miles200) [reply Projects/CoX/Utilities/SEGSAdmin/SEGSAdminTool.h]: Fixed

**Comment by ldilley** [comment]
Post fixes and pending build, this looks good. Great job and excellent persistence, @miles200!

**Pull request merged**